server/src/main/java/org/opensearch/common/util/concurrent/OpenSearchExecutors.java

```diff
@@ -172,6 +172,33 @@ public static OpenSearchThreadPoolExecutor newFixed(
         );
     }
 
+    public static OpenSearchThreadPoolExecutor newResizable(
+        String name,
+        int size,
+        int queueCapacity,
+        ThreadFactory threadFactory,
+        ThreadContext contextHolder
+    ) {
+        BlockingQueue<Runnable> queue;
+        if (queueCapacity < 0) {
+            queue = ConcurrentCollections.newBlockingQueue();
```

Comment:
When do you want this? When I call `newResizable` I don't expect a non-resizable blocking queue. Should this assert/raise?

Reply:
:+1: will throw an exception instead, should be always specified.